internal/go_repository.bzl

```diff
@@ -14,10 +14,29 @@
 
 load("@io_bazel_rules_go//go/private:common.bzl", "env_execute", "executable_extension")
 load("@bazel_gazelle//internal:go_repository_cache.bzl", "read_cache_env")
+load("@bazel_tools//tools/build_defs/repo:utils.bzl", "read_netrc", "use_netrc")
 
 # We can't disable timeouts on Bazel, but we can set them to large values.
 _GO_REPOSITORY_TIMEOUT = 86400
 
+# Inspired by @bazel_tools//tools/build_defs/repo:http.bzl
+def _get_auth(ctx, urls):
```

Comment:
> I expect folks will want to have just one netrc file shared by all rules, so let's add these attributes to `go_repository_config` as well and make sure they can be read by `go_repository` if they're set there.

Reply:
Sync'd offline, and realized this is not feasible because `get_auth` results cannot be passed in as parameters to `go_repository` rule.